client/landing/gutenboarding/stores/domain-suggestions/selectors.ts

```diff
@@ -1,15 +1,65 @@
+/**
+ * External dependencies
+ */
+import { select } from '@wordpress/data';
+
 /**
  * Internal dependencies
  */
+import { STORE_KEY } from '.';
 import { DomainSuggestionQuery } from './types';
 import { State } from './reducer';
-import { getSerializedDomainsSuggestionsQuery } from 'state/domains/suggestions/utils';
+import { stringifyDomainQueryObject } from './utils';
 
 export const getState = ( state: State ) => state;
-export const getDomainSuggestions = ( state: State, queryObject: DomainSuggestionQuery ) => {
-	const serializedQuery = getSerializedDomainsSuggestionsQuery( queryObject );
-	if ( serializedQuery ) {
-		return state.domainSuggestions[ serializedQuery ];
-	}
-	return null;
+
+type DomainSuggestionSelectorOptions = Partial< Exclude< DomainSuggestionQuery, 'query' > >;
+export const getDomainSuggestions = (
+	state: State,
+	search: string,
+	options: DomainSuggestionSelectorOptions = {}
+) => {
+	const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
+
+	// We need to go through the `select` store to get the resolver action
+	return select( STORE_KEY ).__internalGetDomainSuggestions( normalizedQuery );
+};
+
+/**
+ * Do not use this selector. It is for internal use.
+ *
+ * @private
+ */
+export const __internalGetDomainSuggestions = (
+	state: State,
+	queryObject: DomainSuggestionQuery
+) => {
+	return state.domainSuggestions[ stringifyDomainQueryObject( queryObject ) ];
 };
+
+/**
+ * Normalize domain query
+ *
+ * It's important to have a consistent, reproduceable representation of a domains query so that the result can be
+ * stored and retrieved.
+ *
+ * @see client/state/domains/suggestions/utils.js
+ * @see client/components/data/query-domains-suggestions/index.jsx
+ */
+function normalizeDomainSuggestionQuery(
+	search: string,
+	queryOptions: DomainSuggestionSelectorOptions
+): DomainSuggestionQuery {
+	return {
+		// Defaults
+		include_wordpressdotcom: false,
+		quantity: 5,
+		vendor: 'variation2_front',
```

Comment:
This seems _so_ specific (and potentially resulting from a previous A/B test) that I'm thinking maybe we should drop it from the defaults? After all, the endpoint works without an explicitly specified `vendor` :thinking:

Reply:
@klimeryk can I direct that question your way ☝️ 

Should we omit `vendor` by default and assume the server will use the "control" vendor?

I wonder if we'd want to keep the default vendor explicit so that package consumers would be made aware of the vendor change?

Reply:
If we don't specify a default on the client side, we are possibly running into a situation where:

 - the code in Calypso uses the default variation set there explicitly (from an A/B test for example)
 - the code in Gutenboarding uses the default from the endpoint which can be different (hardcoded safe default value to prevent failures, for example)

I could be also missing how this code is supposed to be used - I'm not familiar with how Gutenboarding is supposed to work.

Reply:
> I could be also missing how this code is supposed to be used - I'm not familiar with how Gutenboarding is supposed to work.

Nothing out of the ordinary there -- we'll use it to suggest a list of domains to the user, based upon the site title (and possibly verticals) they provide us with :sweat_smile:

Reply:
Hehe, well, from previous testing that does not covert well, but maybe it will be different in Gutenboarding 🤞 Previously the whole experience was a bit disjointed, hoping Gutenberg ties it all up nicely.